### PR TITLE
[Feature] Migration to v1 resources

### PR DESCRIPTION
**Pull request by erdrix**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | 
| License         | Apache 2.0

### What's in this PR?

Prepare version `1.0.0` with the new version `v1` for stable CRDs. Migrate the package for resource definition and enable the migration webhook.

### Why?

To do things the right way :D 

### Checklist

- [x] Implementation tested
- [X] Error handling code meets the [guideline](docs/error-handling-guide.md)
- [X] Logging code meets the guideline
- [x] User guide and development docs updated (if needed)
- [x] Append changelog with changes

### TODO

- [x] Helm chart: Webhook integration
- [x] Update documentation with v1 resources
- [x] Update sample with v1 resources

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -128,7 +128,8 @@ generate: controller-gen
 # Generate manifests e.g. CRD, RBAC etc.
 .PHONY: manifests
 manifests: controller-gen
-	$(CONTROLLER_GEN) rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+#	$(CONTROLLER_GEN) rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+	$(CONTROLLER_GEN) rbac:roleName=manager-role crd:maxDescLen=0 webhook paths="./..." output:crd:artifacts:config=config/crd/bases
```
Comment (michael81877) [inline Makefile:132]: Why `maxDescLen=0`?

**Comment by michael81877** [comment]
I know this may be a lot to ask, but could we unit test the conversions to make sure we copy entire objects & set defaults (where applicable) appropriately? You've nicely broken down conversion methods into smaller conversion functions, so perhaps it won't be too challenging to test each bit.

Thoughts?

File: go.mod [hunk 2]
```diff
@@ -9,7 +9,6 @@ require (
 	github.com/dgrijalva/jwt-go v3.2.0+incompatible
 	github.com/go-logr/logr v1.2.2
 	github.com/go-logr/zapr v1.2.3
-	github.com/golang-jwt/jwt v3.2.2+incompatible
```
Comment (michael81877) [inline go.mod:12]: I believe this change incidentally re-introduces a vulnerability fixed in this PR #177. We want to keep the `golang-jwt/jwt` package and remove `dgrijalva/jwt-go`.

File: helm/nifikop/README.md [hunk 3]
```diff
@@ -20,28 +20,30 @@ which implements kubernetes custom ressource definition.
...
-| `metrics.enabled`                | deploy service for metrics                                                                                                                                                           | `false`                  |
-| `metrics.port`                   | Set port for operator metrics                                                                                                                                                        | `8081`                   |
-| `debug.enabled`                  | activate DEBUG log level                                                                                                                                                             | `false`                  |
-| `certManager.clusterScoped`      | If true setup cluster scoped resources                                                                                                                                               | `false`                  |
+<<<<<<< HEAD
```
Comment (michael81877) [inline helm/nifikop/README.md:23]: Accidental git merge remnants?

File: main.go [hunk 4]
```diff
@@ -193,8 +199,35 @@ func main() {
 		logger.Error("unable to create controller", zap.String("controller", "NifiNodeGroupAutoscaler"), zap.Error(err))
 		os.Exit(1)
 	}
-	// +kubebuilder:scaffold:builder
 
+	if webhookEnabled {
```
Comment (michael81877) [inline main.go:203]: I think you've missed the `NifiNodeGroupAutoscaler` here.

**Comment by michael81877** [comment]
Was the `NifiNodeGroupAutoscaler` left out of `v1` intentionally?

**Comment by juldrixx** [comment]
> Was the `NifiNodeGroupAutoscaler` left out of `v1` intentionally?

Yes, it was. The functionality is recent, so we keep it as alpha until it is well proven.

**Comment by michael81877** [comment]
> > Was the `NifiNodeGroupAutoscaler` left out of `v1` intentionally?
> 
> Yes, it was. The functionality is recent, so we keep it as alpha until it is well proven.

This sounds reasonable to me. Thanks for explaining!

File: main.go (hunk 4 above)
Comment (erdrix) [reply main.go:203]: Not necessary because we only manage the v1alpha1 version for this resource!

File: Makefile (hunk 1 above)
Comment (erdrix) [reply Makefile:132]: If you apply the CRD, you will get this error:
```console
The CustomResourceDefinition "[nificlusters.nifi.konpyutaika.com](http://nificlusters.nifi.konpyutaika.com/)" is invalid: metadata.annotations: Too long: must have at most 262144 bytes 
```

We add this to the skip description field : https://github.com/kubernetes-sigs/controller-tools/issues/441#issuecomment-650483628

**Comment by erdrix** [comment]
> I know this may be a lot to ask, but could we unit test the conversions to make sure we copy entire objects & set defaults (where applicable) appropriately? You've nicely broken down conversion methods into smaller conversion functions, so perhaps it won't be too challenging to test each bit.
> 
> If you'd like, i can help knock it out.
> 
> Thoughts?

@michael81877 
I'm all for this idea, but to be quite honest, I won't have time to do it... 

If you think you can help on this before the end of the week, that would be great! If not, we can release the version, and add unit tests later :)

**Comment by michael81877** [comment]
Alright i've raised #201 to add unit tests. I actually discovered two bugs that i fixed in the process. These unit tests at least cover happy path for all api conversions.

File: go.mod (hunk 2 above)
Comment (michael81877) [reply go.mod:12]: I corrected this in PR #201

File: Makefile (hunk 1 above)
Comment (michael81877) [reply Makefile:131]: Ah, gotcha. I've seen this a lot in ArgoCD, which forces you to re-create the resource instead of applying. Makes sense!

**Pull request merged**